src/circuit_writer/ir.rs

```diff
@@ -0,0 +1,1119 @@
+use ark_ff::Zero;
```

Comment:
btw as most of this file will be useful in the future to have an IR pass, I think it should be moved to an circ_ir folder or something, and moved out of the circuit_writer